File: Code/Data/TransferSQL/SqlTest/PlantProfile.cs
using System;
using System.Collections.Generic;
using System.Text;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace SqlTest
{
    public class PlantProfile
    {
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }

        [BsonRepresentation(BsonType.Int32)]
        public int PlantID
        {
            get; set;
        }

        [BsonRepresentation(BsonType.String)]
        public string PlantName
        {
            get; set;
        }

        [BsonRepresentation(BsonType.Double)]
        public double Temperature
        {
            get; set;
        }

        [BsonRepresentation(BsonType.Int32)]
        public double Light
        {
            get; set;
        }

        [BsonRepresentation(BsonType.Double)]
        public double CO2
        {
            get; set;
        }

        [BsonRepresentation(BsonType.Double)]
        public double Humidity
        {
            get; set;
        }

        [BsonRepresentation(BsonType.Double)]
        public double AmountOfWater
        {
            get; set;
        }

        [BsonRepresentation(BsonType.Double)]
        public double HoursSinceWatering
        {
            get; set;
        }

        [BsonRepresentation(BsonType.DateTime)]
        public DateTime DateTime
        {
            get; set;
        }

        public PlantProfile(int plantID, string plantName, double temperature,Int32 Light, double CO2, double humidity, double amountOfWater, double hoursSinceWatering, DateTime dateTime)
        {
            this.PlantID = plantID;
            this.PlantName = plantName;
            this.Temperature = temperature;
            this.Light = Light;
            this.CO2 = CO2;
            this.Humidity = humidity;
            this.AmountOfWater = amountOfWater;
            this.HoursSinceWatering = hoursSinceWatering;
            this.DateTime = dateTime;
        }


    }
}

File: Code/Embedded/IOT/BridgeApp/src/model/PlantProfile.java
package model;

import com.fasterxml.jackson.core.JsonProcessingException;

import java.io.IOException;

/**
 * DTO for Plants
 */
public class PlantProfile {
	public int PlantID;
	public String PlantName;
	public double Temperature;
	public double CO2;
	public double Humidity;
	public double AmountOfWater;
	public double HoursSinceWatering;
	public int Light;

	public String toJson() {
		try {
			return JsonConverter.objectMapper.writeValueAsString(this);
		} catch(JsonProcessingException e) {
			return null;
		}
	}

	public static PlantProfile fromJson(String json) {
		try {
			return JsonConverter.objectMapper.readValue(json, PlantProfile.class);
		} catch(IOException e) {
			return null;
		}
	}
}
